### PR TITLE
feat(chart): add annotations to apiserver service; add tolerations and nodeSelector to job.

### PR DESCRIPTION
**Pull request by EnnnOK**
**What type of PR is this?**
update chart

**What this PR does / why we need it**:
1. Chart add annotations to apiserver service
2. Chart add tolerations and nodeSelector to pre-install/post-install/post-delete jobs

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Poor12** [comment]
Hi @EnnnOK , thank you for your first PR! We appreciate your contribution and effort, we really value your work and hope you will continue to participate in our project.
Please sign the DCO first.

**Comment by EnnnOK** [comment]
done

**Review by RainbowMango — COMMENTED** [review]
Thanks.
/assign @Poor12

File: charts/karmada/templates/karmada-apiserver.yaml [hunk 1]
```diff
@@ -154,6 +154,10 @@ metadata:
   namespace: {{ include "karmada.namespace" . }}
   labels:
   {{- include "karmada.apiserver.labels" . | nindent 4 }}
+  {{- with .Values.apiServer.serviceAnnotations }}
```
Comment (Poor12) [inline charts/karmada/templates/karmada-apiserver.yaml:157]: I am curious why you need an annotation in service. Is there any special use cases?

**Review by Poor12 — COMMENTED** [review]
Could you update the README together? LINK: https://github.com/karmada-io/karmada/tree/master/charts/karmada#common-parameters

**Comment by EnnnOK** [comment]
> Could you update the README together? LINK: https://github.com/karmada-io/karmada/tree/master/charts/karmada#common-parameters

Updated.

**Comment by Poor12** [comment]
Thanks @EnnnOK . Could you merge 2 commits to one so that we can move this pr forward? You can use git rebase -i HEAD~2 to do this job.

**Comment by RainbowMango** [comment]
Here is the guideline for squash commits: https://karmada.io/docs/contributor/github-workflow#squash-commits.

**Comment by EnnnOK** [comment]
> Thanks @EnnnOK . Could you merge 2 commits to one so that we can move this pr forward? You can use `git rebase -i HEAD~2` to do this job.

Squashed

**Comment by EnnnOK** [comment]
> Here is the guideline for squash commits: https://karmada.io/docs/contributor/github-workflow#squash-commits.

OK

**Comment by Poor12** [comment]
/lgtm
Need @RainbowMango to run CI workflow.

**Review by RainbowMango — COMMENTED** [review]
Done

**Pull request merged**